src/book/03-guides/01-main-codebase/03-recording-playback.mdx

```diff
@@ -0,0 +1,120 @@
+---
+section: Guides
+chapter: Main Codebase
+title: Recording and Playback
+description: Recording data and playing it back.
+slug: /guides/main/recording-playback
+---
+
+NUbots data is stored in [NUClear Binary Stream (NBS) files](/system/foundations/nuclear#nbs-files). Any [messages](https://github.com/NUbots/NUbots/tree/master/shared/message) that are passed around the NUbots system can be recorded.
+
+## Recording Data
+
+The following steps detail how to record data. An example will be given throughout involving recording sensors messages.
+
+1. Set up the role. Most of the time this involves making minor changes to an existing role.
+
+   1. The role must contain the module/s which emit the message/s you want to record.
+   2. The role must contain the module `support::logging::DataLogging`.
+
+   For example, you may want to record [Sensors](https://github.com/NUbots/NUbots/blob/master/shared/message/input/Sensors.proto) messages. You could use the existing [sensortest](https://github.com/NUbots/NUbots/blob/master/roles/sensortest.role) role and add `support::logging::DataLogging` to it.
+
+2. Set up the configuration file. This involves making changes to the `DataLogging.yaml` file. Add in the message you want to log, specifying `true`.
+
+   In this example, the sensors message already exists in the file (`message.input.Sensors`). Set the boolean to `true` next to this message.
+
+   If you would like to log a different message:
+
+   1. Find the `.proto` file that defines that message, such as `Sensors.proto`.
+   2. Find what the `package` is. For `Sensor.proto`, it is `package message.input`.
+   3. Piece this together with the message name (`message Sensors` in the `.proto` file) to get `message.input.Sensors`
+   4. Add this to the configuration file with the other messages.
+
+   You can enable multiple messages at once.
+
+3. Build the code and install it to a robot. To find out how to do this, visit the [Getting Started](/guides/nubots/getting-started) page.
+
+4. Run the binary. In our example, this would be `./sensortest`.
+
+5. Stop the binary when you have finished recording by pressing <kbd>CTRL</kbd> + <kbd>c</kbd>.
+
+6. Move the newly created NBS file to a computer for safe storage.
+   1. The NBS file will be located on the robot in the `log/role_name/` folder. The name will correspond to the time and date it was created, according to the robot's clock. Make note of the name. In our example, this may be `log/sensortest/18072020T082100.nbs`.
+   2. Copy the file across. On the destination computer, run `scp nubots@<address>:log/role_name/file_name path/to/destination`. In our example, we may want to run `scp nubots@<address>:log/sensortest/18072020T082100.nbs .`. The file will then be copied to the folder you ran the command in. `<address>` corresponds to the address of the robot.
```

Comment:
It could be helpful for us poor windows peasants to have descriptions of what the commands are and what they are doing. Or are we assuming that by this point, people have a bit of a grasp of the command line?